### PR TITLE
perf: Calculate length of name column in perf tests

### PR DESCRIPTION
**Pull request by StephanDollberg**
Auto caluclate the maximum needed size of the name column in the perf
tests.

This avoids garbled output like the following:

```
stephan@rp:/build/seastar/build/release$ ./tests/perf/rpc_perf -i 10 --md-output /tmp/seastar.md 2>&1  | grep -v -i -E "reactor stalled|kernel callstack"
INFO  2024-02-21 12:18:01,631 seastar - Reactor backend: io_uring
single run iterations:    10
single run duration:      1.000s
number of runs:           5
number of cores:          32
random seed:              62032321

test                                      iterations      median         mad         min         max      allocs       tasks        inst
lz4.small_random_buffer_compress                  10   151.400ns     0.100ns   150.800ns   151.500ns       1.000       0.000      2198.0
lz4.small_zeroed_buffer_compress                  10    96.100ns     0.800ns    95.300ns   531.300ns       1.000       0.000       785.0
lz4.large_random_buffer_compress                  10     2.489ms    34.602us     2.454ms     2.576ms     146.000       0.000   3239849.6
lz4.large_zeroed_buffer_compress                  10     1.082ms    20.812us     1.061ms     1.118ms       9.000       0.000  16876805.2
lz4.small_random_buffer_decompress                10    33.500ns     0.700ns    32.500ns    35.200ns       1.000       0.000       354.0
lz4.small_zeroed_buffer_decompress                10    48.000ns     1.500ns    46.500ns    50.200ns       1.000       0.000       520.0
lz4.large_random_buffer_decompress                10     2.405ms     9.514us     2.395ms     2.421ms     145.000       0.000   2375950.1
lz4.large_zeroed_buffer_decompress                10     2.045ms     2.414us     1.988ms     2.047ms     136.000       0.000  13095442.0
lz4_fragmented.small_random_buffer_compress          10   155.400ns     2.000ns   152.100ns   157.400ns       1.000       0.000      2678.0
lz4_fragmented.small_zeroed_buffer_compress          10    91.900ns     1.200ns    90.200ns    93.700ns       1.000       0.000       780.0
lz4_fragmented.large_random_buffer_compress          10     4.111ms     8.553us     4.098ms     4.153ms     188.000       0.000  26707588.2
lz4_fragmented.large_zeroed_buffer_compress          10   494.053us     1.142us   488.865us   506.775us      10.000       0.000  17550547.9
lz4_fragmented.small_random_buffer_decompress          10    37.400ns     0.300ns    37.100ns    39.000ns       1.000       0.000       432.0
lz4_fragmented.small_zeroed_buffer_decompress          10    49.500ns     0.200ns    47.900ns    51.700ns       1.000       0.000       598.0
lz4_fragmented.large_random_buffer_decompress          10   654.409us    11.390us   630.640us   665.800us     274.000       0.000    700311.3
lz4_fragmented.large_zeroed_buffer_decompress          10   753.631us     4.280us   747.277us   759.924us     265.000       0.000  14821658.5
```

Instead we now preserve the properly formatted table output:

```
$ ./tests/perf/rpc_perf -i 10 --md-output /tmp/seastar.md 2>&1  | grep -v -i -E "reactor stalled|kernel callstack"
INFO  2024-02-21 12:15:41,140 seastar - Reactor backend: io_uring
single run iterations:    10
single run duration:      1.000s
number of runs:           5
number of cores:          32
random seed:              2089348534

test                                           iterations      median         mad         min         max      allocs       tasks        inst
lz4.small_random_buffer_compress                       10   150.900ns     1.700ns   147.100ns   155.200ns       1.000       0.000      2198.0
lz4.small_zeroed_buffer_compress                       10    96.200ns     1.900ns    94.300ns   101.200ns       1.000       0.000       785.0
lz4.large_random_buffer_compress                       10     2.425ms    34.466us     2.387ms     2.468ms     146.000       0.000   3240088.3
lz4.large_zeroed_buffer_compress                       10     1.070ms    14.063us     1.052ms     1.112ms       9.000       0.000  16876805.3
lz4.small_random_buffer_decompress                     10    33.300ns     0.900ns    31.700ns    35.400ns       1.000       0.000       354.0
lz4.small_zeroed_buffer_decompress                     10    46.200ns     0.300ns    45.900ns    47.200ns       1.000       0.000       520.0
lz4.large_random_buffer_decompress                     10     2.444ms    12.680us     2.385ms     2.457ms     145.000       0.000   2375950.1
lz4.large_zeroed_buffer_decompress                     10     2.056ms    23.872us     2.021ms     2.138ms     136.000       0.000  13095441.7
lz4_fragmented.small_random_buffer_compress            10   152.900ns     0.400ns   152.200ns   153.600ns       1.000       0.000      2678.0
lz4_fragmented.small_zeroed_buffer_compress            10    93.100ns     1.500ns    90.300ns    94.600ns       1.000       0.000       780.0
lz4_fragmented.large_random_buffer_compress            10     4.096ms    14.196us     4.060ms     4.110ms     188.000       0.000  26707284.8
lz4_fragmented.large_zeroed_buffer_compress            10   504.067us     3.066us   496.572us   507.133us      10.000       0.000  17550548.1
lz4_fragmented.small_random_buffer_decompress          10    37.100ns     0.400ns    36.100ns    39.300ns       1.000       0.000       432.0
lz4_fragmented.small_zeroed_buffer_decompress          10    48.300ns     0.200ns    48.100ns    51.200ns       1.000       0.000       598.0
lz4_fragmented.large_random_buffer_decompress          10   622.140us     3.974us   618.166us   648.211us     274.000       0.000    700310.9
lz4_fragmented.large_zeroed_buffer_decompress          10   750.012us     1.575us   747.780us   756.675us     265.000       0.000  14821658.6
```

We do the same for the markdown output. While the rendered output is the
same it looks cleaner when the markdown is not rendered (e.g.: in commit
messages).

### CODE REVIEW

File: tests/perf/perf_tests.cc [hunk 1]
```diff
@@ -201,14 +210,25 @@ struct stdout_printer final : result_printer {
...
+private:
+  std::string header_format_string() const {
+    return fmt::format("{{:<{}}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}}\n",
+        name_column_length());
+  }
+
+  std::string format_string() const {
+    return fmt::format("{{:<{}}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11.3f}} {{:>11.3f}} {{:>11.1f}}\n",
+        name_column_length());
+  }
```
Comment (avikivity) [inline tests/perf/perf_tests.cc:231]: You can write `format("{:<{}}", name, width)` to avoid the awkward runtime format strings.

Comment (StephanDollberg) [reply tests/perf/perf_tests.cc]: Oh nice, yes that's a lot cleaner. Have changed.

File: tests/perf/perf_tests.cc [hunk 2]
```diff
@@ -376,7 +390,13 @@ void run_all(const std::vector<std::string>& tests, const config& conf)
         return tests.empty() || it != tests.end();
     };
 
+    size_t max_name_column_length = 0;
+    for (auto&& test : all_tests() | boost::adaptors::filtered(can_run)) {
+        max_name_column_length = std::max(max_name_column_length, test->name().size());
+    }
```
Comment (tchaikov) [inline tests/perf/perf_tests.cc:396]: how about using `std::ranges::max_element()` ? more à la mode this way. as we have the luxury of using C++20 now..

Comment (StephanDollberg) [reply tests/perf/perf_tests.cc:396]: I was using the non-ranges version at first but it's not actually much nicer or shorter as it returns an iterator so you still kind of have to do a check afterwards whether the iterator is valid (i.e.: empty range).

**Review by tchaikov — APPROVED** [review]
lgtm.

**Pull request merged**